cmd/pitr/recoverer/recoverer.go

```diff
@@ -0,0 +1,252 @@
+package recoverer
+
+import (
+	"bytes"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/percona/percona-xtradb-cluster-operator/cmd/pitr/storage"
+	"github.com/pkg/errors"
+)
+
+type Recoverer struct {
+	recoverTime      int64
+	storage          *storage.S3
+	localDest        string
+	pxcUser          string
+	pxcHost          string
+	pxcPass          string
+	recoverType      RecoverType
+	pxcServiceName   string
+	binlogs          []string
+	backupNAme       string
+	excludingGTIDSet string
+}
+
+type Config struct {
+	PXCServiceName string
+	PXCUser        string
+	PXCPass        string
+	S3Endpoint     string
+	S3AccessKeyID  string
+	S3AccessKey    string
+	S3BucketName   string
+	S3Region       string
+	RecoverTime    int64
+	RecoverType    string
+	BackupName     string
+}
+
+type RecoverType string
+
+func New(c Config) (*Recoverer, error) {
+	s3, err := storage.NewS3(c.S3Endpoint, c.S3AccessKeyID, c.S3AccessKey, c.S3BucketName, c.S3Region, true)
+	if err != nil {
+		return nil, errors.Wrap(err, "new storage manager")
+	}
+	return &Recoverer{
+		storage:        s3,
+		recoverTime:    c.RecoverTime,
+		pxcUser:        c.PXCUser,
+		pxcPass:        c.PXCPass,
+		pxcServiceName: c.PXCServiceName,
+		recoverType:    RecoverType(c.RecoverTime),
+		backupNAme:     c.BackupName,
+	}, nil
+}
+
+const (
+	Latest      RecoverType = "latest"
+	Date        RecoverType = "date"
+	Transaction RecoverType = "transaction"
+	Skip        RecoverType = "skip"
+)
```

Comment:
would be nice to have any explanatory comments to these types

Reply:
https://github.com/percona/percona-xtradb-cluster-operator/pull/673/commits/74347e94f05b3b59a0399735a7a9e3ea4ae3d768